src/openms/source/SYSTEM/File.cpp

```diff
@@ -814,12 +814,13 @@ namespace OpenMS
   {
   }
 
-  const String& File::TemporaryFiles_::newFile()
+  String File::TemporaryFiles_::newFile()
   {
     String s = getTempDirectory().ensureLastChar('/') + getUniqueName();
     std::lock_guard<std::mutex> _(mtx_);
     filenames_.push_back(s);
-    return filenames_.back();
```

Comment:
Why is this not handled by the lock?

Reply:
because it returned a **reference**.
The function ends, the lock is released, but the reference still exists. Then comes another thread and reallocates the vector. Reference becomes invalid and boom...

Reply:
ahhh right of course